skaha/src/main/java/org/opencadc/skaha/utils/CommonUtils.java

```diff
@@ -24,10 +27,46 @@ public static String encodeBase64(final byte[] encodedValue) {
      * Obtain the first configured Service URI for the given base standard ID.
      *
      * @param baseStandardID The URI to lookup.
-     * @return A single URI (first matching).  Never null.
+     * @return A single URI (first matching). Never null.
      */
     public static URI firstLocalServiceURI(final URI baseStandardID) {
         final Set<URI> serviceURIs = new LocalAuthority().getServiceURIs(baseStandardID);
         return serviceURIs.stream().findFirst().orElseThrow(IllegalStateException::new);
     }
+
+    /**
+     * Format to Binary SI values.
+     *
+     * @param bytes Bytes reported from Kubernetes.
+     * @return String formatted, never null.
+     */
+    public static String formatMemoryFromBytes(final BigDecimal bytes) {
+        final long longBytes = bytes.longValue();
+        final long absB = longBytes == Long.MIN_VALUE ? Long.MAX_VALUE : Math.abs(longBytes);
+
+        if (absB < 1024) {
+            return longBytes + " B";
+        }
+
+        long value = absB;
+        CharacterIterator ci = new StringCharacterIterator("KMGTPE");
+        for (int i = 40; i >= 0 && absB > 0xfffccccccccccccL >> i; i -= 10) {
```

Comment:
Better docs for this magic hex bit-shifting?

Reply:
Made readable...